lib/record_store/provider/ns1/patch_api_header.rb

```diff
@@ -0,0 +1,21 @@
+require 'net/http'
+
+# Patch the method which retrieves headers for API rate limit dynamically
+module NS1::Transport
+  class NetHttp
+    def process_response(response)
+      sleep(response.to_hash["x-ratelimit-period"].first.to_i /
```

Comment:
Have you confirmed that this prevents the lack of precision in this division?  Consider looking at the implementation of `Integer./` to add context to @XaF 's comment.

Reply:
I have confirmed that this prevents the lack of precision in this devision. I actually left more context in @xaf's  comment but I decided not to.

Before I put, `to_f`. 300 (period) / 899 (remaining) was 0.
Now it returns 0.33..... 

But if you want me to handle in case headers aren't valid integer or headers aren't integer, I can consider looking at the `Integer.`.

Reply:
Gotcha 👍 .  Had a bit of a brain freeze there.